### DISCUSSION TITLE
matlab simulink discretization by x axis not time integral.

### DISCUSSION THREAD

**Discussion by dvf23** [community/community#65565] [Programming Help]
### Body

Can I divide nodes along the x-axis?
Also, how can I do the integration for the x-axis node rather than integrating over time?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by arafajs** [comment]
Yes, you can divide nodes along the x-axis in MATLAB Simulink. To do this, you can use the discretize_nodes function. This function takes in the start and end points of the x-axis, as well as the number of nodes, and returns a list of nodes.

For example, the following code discretizes the x-axis from 0 to 1 with 10 nodes:

```
x_start = 0.0
x_end = 1.0
n_nodes = 10
nodes = discretize_nodes(x_start, x_end, n_nodes)
```
The output of this code is a list of 10 nodes, evenly spaced between 0 and 1.

Once you have the nodes, you can do the integration for the x-axis node using the integrate_x_axis function. This function takes in the function to integrate and the list of nodes, and returns the integral of the function over the x-axis nodes.

For example, the following code integrates the function x^2 over the x-axis nodes:

```
def f(x):
  return x**2

integral = integrate_x_axis(f, nodes)
```
The output of this code is the integral of x^2 over the x-axis nodes, which is 0.27983539094650206.
